shopinvader_payment_gift_card/services/cart.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (C) 2022 Akretion (<http://www.akretion.com>).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import date
+
+from odoo.addons.base_rest.components.service import to_bool
+from odoo.addons.component.core import Component
+
+
+def to_date(value):
+    return date.fromisoformat(value)
+
+
+class CartService(Component):
+    _inherit = "shopinvader.cart.service"
+
+    def _convert_one_line(self, line):
+        if line.gift_card_id:
+            line.price_unit = line.gift_card_id.initial_amount
+        res = super()._convert_one_line(line)
+        res.update(self._get_gift_card_id(line))
+        return res
+
+    def _get_gift_card_id(self, line):
+        return {"gift_card_id": line.gift_card_id}
```

Comment:
Shouldn't this be `line.gift_card_id.id`?
I'm getting `TypeError: Object of type gift.card is not JSON serializable` when accessing a cart without gift card

Reply:
Indeed, I changed it.